File: 118.pascals-triangle.cpp
class Solution
{
public:
    vector<vector<int>> generate(int numRows)
    {
        int i = 1;
        vector<vector<int>> ans;
        vector<int> a;
        a.push_back(1);
        ans.push_back(a);
        while (i < numRows)
        {
            ans.push_back(makePascle(ans[i - 1]));
            i++;
        }
        return ans;
    }
    vector<int> makePascle(vector<int> num)
    {
        vector<int> ans;
        ans.push_back(1);
        for (int i = 0; i < num.size() - 1; i++)
            ans.push_back(num[i] + num[i + 1]);
        ans.push_back(1);
        return ans;
    }
};

File: 118.pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = []
        ans.append([1])
        if numRows == 1:
            return ans

        k = 1
        while k < numRows:
            arr = [1]
            for i in range(1,k,1):
                arr.append(ans[k-1][i-1]+ans[k-1][i])
            arr.append(1)
            ans.append(arr)
            k+=1
        return ans